site/templates/items/tagline.jet

```diff
@@ -9,85 +13,56 @@
-    {{end}}
-    {{if isset(.Episodes)}}
-      {{dividers = dividers + 1}}
-    {{end}}
-    {{if isset(.ReleaseDate) && .ReleaseDate.Year() > 1}}
-      {{dividers = dividers + 1}}
-    {{end}}
-    {*{ There should be one less dividers than tagline items. }*}
-    {{dividers = dividers - 1}}
-    
-    {{if classification}}
-      <s72-classification-label data-slug="{{.Slug}}" data-layout="tooltip"></s72-classification-label>
-    {{end}}
+    {{showDivider := false}}
 
     {{if isset(.Runtime)}}
-      {{yield taglineItem() content}}
+      {{yield item(class="runtime") content}}
         {{if .Runtime > 60}}
-          {{.Runtime.Hours()}}{{i18n("runtime_hours")}} {{.Runtime.Minutes()}}{{i18n("runtime_minutes")}}
-        {{else}}
-          {{.Runtime}}{{i18n("runtime_minutes")}}
+          {{.Runtime.Hours()}}{{i18n("runtime_hours")}}
         {{end}}
+        {{.Runtime.Minutes()}}{{i18n("runtime_minutes")}}
       {{end}}
-      {{yield taglineDivider(dividers=dividers)}}
-      {{dividers = dividers - 1}}
+      {{showDivider = true}}
     {{end}}
 
     {{if isset(.Items)}}
-      {{yield taglineItem() content}}
+      {{yield item(class="items") content}}
         {{i18n("bundle_items_all_films", len(.Items))}}
       {{end}}
-      {{yield taglineDivider(dividers=dividers)}}
-      {{dividers = dividers - 1}}
+      {{showDivider = true}}
     {{end}}
 
     {{if isset(.Episodes)}}
-      {{yield taglineItem() content}}
+      {{yield item(class="episodes") content}}
         {{i18n("episode_count", len(.Episodes))}}
       {{end}}
-      {{yield taglineDivider(dividers=dividers)}}
-      {{dividers = dividers - 1}}
+      {{showDivider = true}}
     {{end}}
 
     {{if genres && len(itemGenres) > 0}}
       {{if genresLimit > -1 && len(itemGenres) > genresLimit}}
         {{itemGenres = itemGenres[:genresLimit]}}
       {{end}}
-      {{yield taglineItem() content}}
+      {{yield item(class="genres") content}}
         {{itemGenres}}
       {{end}}
-      {{yield taglineDivider(dividers=dividers)}}
-      {{dividers = dividers - 1}}
+      {{showDivider = true}}
     {{end}}
 
-    {{if isset(.ReleaseDate) && .ReleaseDate.Year() > 1}}
-      {{yield taglineItem() content}}
+    {{if isset(.ReleaseDate) && .ReleaseDate.Year() > 1888}}
+      {{yield item(class="release-date") content}}
         {{.ReleaseDate.Year()}}
       {{end}}
-      {{yield taglineDivider(dividers=dividers)}}
-      {{dividers = dividers - 1}}
+      {{showDivider = true}}
     {{end}}
   </div>
 {{end}}
 
-{{block taglineItem()}}
-  <span class="meta-item-tagline-item">{{yield content}}</span>
-{{end}}
-
-{{block taglineDivider(dividers)}}
-  {{if dividers > 0}}
-    <span class="meta-item-tagline-divider">•</span>
-  {{end}}
+{{block item(class="missing", showDivider=showDivider)}}
+  <span class="{{class}}">
+    {{if showDivider}}
+      <span class="divider">•</span>
+    {{end}}
+    <span class="item">{{yield content}}</span>
+  </span>
```

Comment:
This changes the DOM output from:

```html
<item1><divider><item2><divider><item3>
```

... to:

```html
<wrapper><item1></wrapper>
<wrapper><divider><item2></wrapper>
<wrapper><divider><item3></wrapper>
```

Reply:
![image](https://user-images.githubusercontent.com/2253814/169017076-b84780e2-dd3b-48c5-82ab-fa4bdd7e0d43.png)

Reply:
Note the whitespace around `item`'s - that's pre-existing. We can remove whitespace in the jet so it doesn't appear in the final output but it makes the code less readable so left as is. 

There is [whitespace trimming](https://github.com/CloudyKit/jet/blob/master/docs/syntax.md#whitespace-trimming) that could potentially make use of but doesn't seem to available in the currently used version - something for the v2 Jet upgrade.

Reply:
I think the DOM structure gives good CSS targets.

Reply:
It gives good CSS targets for Bryny's goal (hide an item and a divider with one selector), but isn't this a step back from the flat DOM structure for margins between everything?

To put the margins back in, you would need to do something like:
```
.meta-item-tagline {
  > * {
    &:not(:last-child, :only-child, s72-classification-label) {
      margin-right: map-get($spacers, 1); // Currently 0.25rem
    }

    .divider {
      margin-right: map-get($spacers, 1); // Currently 0.25rem
    }
  }
}
```
Which is gross... though only a bit grosser than the existing CSS.

But if we keep the existing flat DOM structure, we can refactor the CSS with `gap` and it will be much cleaner and easier to work on in the future.

Reply:
Heh too smart for me lol.  👍  for margins on dividers only!